### PR TITLE
Introduce Picture in Picture

### PR DESCRIPTION
**Pull request by zigavehovec**
## Description

This PR introduces Picture in Picture handling on Android including a sample usage.

Prework PRs: 
https://github.com/bitmovin/bitmovin-player-flutter/pull/85
https://github.com/bitmovin/bitmovin-player-flutter/pull/86

## Changes

- Add and use the `FlutterPictureInPictureHandler`
- Relay PiP related events
- Basic PiP sample

## Checklist (for PR submitters and reviewers)
- [x] 🗒 `CHANGELOG.md` entry for new/changed features, bug fixes or important code changes
- [ ] 🧪 Tests added and/or updated
- [ ] 📢 New public API is fully documented

### CODE REVIEW

File: android/src/main/kotlin/com/bitmovin/player/flutter/FlutterPlayerView.kt [hunk 1]
```diff
@@ -119,6 +141,9 @@ class FlutterPlayerView(
         sink = null
     }
 
+    private val isPictureInPictureSupported: Boolean
+        get() = Build.VERSION.SDK_INT >= Build.VERSION_CODES.O
+
```
Comment (strangesource) [inline android/src/main/kotlin/com/bitmovin/player/flutter/FlutterPlayerView.kt:146]: NIT: could be a top level function in the file to make it clear that it does not need any state from the view.

File: example/lib/pages/picture_in_picture.dart [hunk 2]
```diff
@@ -60,8 +78,25 @@ class _PictureInPictureState extends State<PictureInPicture> {
     super.dispose();
   }
 
+  // Since PiP on Android is basically just the whole screen fitted in a small
```
Comment (strangesource) [inline example/lib/pages/picture_in_picture.dart:81]:
```suggestion
  // Since PiP on Android is basically just the whole activity fitted in a small
```

**Review by strangesource — APPROVED** [review]
Nice work!

**Review by hawk23 — APPROVED** [review]
Nice work 💪 

- I see that the `PictureInPictureAvailabilityChanged` event is not implemented for Flutter. As it's Android only, do you think it would make sense to expose it?
- As discussed on Slack, when `_player.loadSourceConfig(_sourceConfig);` is commented out in the PiP sample (no source loaded into the player), PiP does not work anymore on my personal Android device. The floating window does not appear and the player view just goes to Fullscreen mode.

File: example/lib/pages/picture_in_picture.dart [hunk 3]
```diff
@@ -34,6 +37,21 @@ class _PictureInPictureState extends State<PictureInPicture> {
...
+      _isInPictureInPicture = true;
+    });
+  }
+
+  void _onPictureInPictureExitEvent(Event event) {
+    _onEvent(event);
+    setState(() {
+      _isInPictureInPicture = false;
+    });
+  }
```
[the remark this answers is not in the document]
Comment (zigavehovec) [reply example/lib/pages/picture_in_picture.dart:54]: Totally agree and as you mentioned there currently is no good way of going around that (at least I couldn't find it).

**Comment by zigavehovec** [comment]
> I see that the PictureInPictureAvailabilityChanged event is not implemented for Flutter. As it's Android only, do you think it would make sense to expose it?

I actually missed this event. I'll create a follow up ticket and we can tackle this in another PR 🙂

> As discussed on Slack, when _player.loadSourceConfig(_sourceConfig); is commented out in the PiP sample (no source loaded into the player), PiP does not work anymore on my personal Android device. The floating window does not appear and the player view just goes to Fullscreen mode.

I looked into this and found some limitation in the native SDK `DefaultPictureInPictureHandler`. We discussed it with @strangesource and decided for a workaround until we improve the situation on the native side. https://github.com/bitmovin/bitmovin-player-flutter/pull/87/commits/edce49e1ae8664ea9b4282907de88757e1540396

File: android/src/main/kotlin/com/bitmovin/player/flutter/FlutterPlayerView.kt (hunk 1 above)
Comment (zigavehovec) [reply android/src/main/kotlin/com/bitmovin/player/flutter/FlutterPlayerView.kt]: https://github.com/bitmovin/bitmovin-player-flutter/pull/87/commits/4f1efd2a592845b1c63e451e62a6905b4e2a20b5

File: example/lib/pages/picture_in_picture.dart (hunk 3 above)
[the remark this answers is not in the document]
Comment (hawk23) [reply example/lib/pages/picture_in_picture.dart:54]: Yeah, I also don't have a better solution right now 👀

**Review by hawk23 — APPROVED** [review]

**Pull request merged**